client/api/src/execution_extensions.rs

```diff
@@ -62,6 +62,18 @@ impl Default for ExecutionStrategies {
 	}
 }
 
+/// Generate the starting set of ExternalitiesExtensions based upon the given capabilities
+pub trait ExtensionsMaker: Send + Sync {
```

Comment:
I don't like `Maker` suffix that much, I would either call it `ExtensionsFactory` or `ExtensionsProvider` or just traitify `Extensions` itself and just have `fn new` function in that trait.

Reply:
Sounds german :D I would also go with `Factory` or `Provider`.

Reply:
`Factory` and `Provider` both sound like more than it does, but sure I don't care what it is named.